### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by alongalky**
Making the readme a bit more welcoming.

### CODE REVIEW

File: readme.MD [hunk 1]
```diff
@@ -1,16 +1,47 @@
 # Dialogs Framework
 The Dialogs Framework is a Python library for writing resumable functions with persistent execution state.
```
Comment (iddan) [inline readme.MD]:
```suggestion
A Python library for creating resumable dialog interfaces
```

File: readme.MD [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
 
-# Examples
+## Resumable what now?
+This framework is intended for long-lived backend flows, such as chatbot conversations. It allows one to write code as a chronological flow, and provides the magic to make execution resume where it left off.
 
-## Set up
+## Code sample
```
Comment (iddan) [inline readme.MD]:
```suggestion
## Example
```

File: readme.MD [hunk 3]
```diff
@@ -1,16 +1,47 @@
...
+        return f"{name} is a beautiful name!"
+    else:
+        return "What is your name?"
+
+# With the dialogs framework
+@dialog()
+def ask_a_name_dialog():
+    run(send_message("What is your name?"))
+    name = run(get_client_response())
+    run(send_message(f"{name} is a beautiful name!"))
```
Comment (iddan) [inline readme.MD]: Show the usage of the framework first.

File: readme.MD [hunk 4]
```diff
@@ -1,16 +1,47 @@
...
+## Code sample
+```python
+# No dialog, manual state manipulation
+def ask_a_name(client_answer, state):
+    if not state.did_ask_already:
+        state.did_ask_already = True
+        name = client_answer
+        return f"{name} is a beautiful name!"
+    else:
+        return "What is your name?"
```
Comment (iddan) [inline readme.MD:25]: In a later section you can show how would it work without the framework and explain the drawbacks

File: readme.MD [hunk 5]
```diff
@@ -1,16 +1,47 @@
...
+def ask_a_name_dialog():
+    run(send_message("What is your name?"))
+    name = run(get_client_response())
+    run(send_message(f"{name} is a beautiful name!"))
+```
+
+## Philosophy
+* No direct access to flow state.
+* No breaks in the flow with `return` statements.
+* No in-memory state, scale-out friendly.
```
Comment (iddan) [inline readme.MD]: You should explain what flow is

File: readme.MD [hunk 6]
```diff
@@ -1,16 +1,47 @@
 # Dialogs Framework
 The Dialogs Framework is a Python library for writing resumable functions with persistent execution state.
 
-# Examples
+## Resumable what now?
+This framework is intended for long-lived backend flows, such as chatbot conversations. It allows one to write code as a chronological flow, and provides the magic to make execution resume where it left off.
```
Comment (iddan) [inline readme.MD:5]: Give an explanation for what flow is. I would also explain how it is being used in K.

File: readme.MD (hunk 1 above)
Comment (alongalky) [reply readme.MD]: Accepted but rephrased, I prefer not to use "dialog" in the definition

File: readme.MD (hunk 3 above)
Comment (alongalky) [reply readme.MD]: Switched 👍

File: readme.MD (hunk 4 above)
Comment (alongalky) [reply readme.MD:25]: I want to keep them close together, I think seeing just the framework code isn't clear enough to the uninitiated.

File: readme.MD (hunk 5 above)
Comment (alongalky) [reply readme.MD]: Removed it from the first point, not sure how to simplify the second

File: readme.MD (hunk 6 above)
Comment (alongalky) [reply readme.MD:5]: I added a paragraph but I couldn't write something clear enough about the K usage, I think it still sounds mysterious to non-users

**Pull request merged**